From 5becb64d08ceca9d7596496283112d7db44cc9e3 Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Tue, 13 Oct 2015 16:02:24 +0800
Subject: [PATCH] Add an IOCTL and a struct for update exposure

Add an IOCTL interface and a structure for getting real exposure value

Change-Id: Ibc1ba7efca18df2cae42b919c6a50df391715575
Original-Tracked-On: #H1504065822
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7449
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
---
 include/linux/atomisp.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/linux/atomisp.h b/include/linux/atomisp.h
index c12d9aa..12640b9 100644
--- a/include/linux/atomisp.h
+++ b/include/linux/atomisp.h
@@ -931,6 +931,13 @@ struct atomisp_acc_state {
 	unsigned int fw_handle;
 };
 
+struct atomisp_update_exposure {
+	unsigned int gain;
+	unsigned int digi_gain;
+	unsigned int update_gain;
+	unsigned int update_digi_gain;
+};
+
 /*
  * V4L2 private internal data interface.
  * -----------------------------------------------------------------------------
@@ -1220,6 +1227,10 @@ struct atomisp_sensor_ae_bracketing_lut {
 
 #define ATOMISP_IOC_S_SENSOR_RUNMODE \
 	_IOW('v', BASE_VIDIOC_PRIVATE + 48, struct atomisp_s_runmode)
+
+#define ATOMISP_IOC_G_UPDATE_EXPOSURE \
+	_IOWR('v', BASE_VIDIOC_PRIVATE + 49, struct atomisp_update_exposure)
+
 /*
  * Reserved ioctls. We have customer implementing it internally.
  * We can't use both numbers to not cause ABI conflict.
-- 
1.9.1

